project/resources/js/Components/Pagination.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+  <div style="text-align:center;">
+    <div class="custom-pagination" v-if="shouldShowPagination">
+      <div class="row center">
+        <div class="col-xs-12 col-sm-3 col-md-3">
+          <label
+            class="col-xs-12 col-md-6 btn btn-light"
+            :disabled="!enabledPrevPageButton"
+            @click="fetchPrevPage" >
+            Anterior
+          </label>
+        </div>
+
+        <div class="col-xs-12 col-sm-5 col-md-6">
+          <label class="btn btn-light"
+                 v-for="button in paginationButtons"
+                 :ref="'paginationButton' + button.page"
+                 :class="{'active': button.active, 'disabled': button.disabled}"
+                 @click="changePage(button.page)" >
+            {{button.text}}
+          </label>
+        </div>
+
+        <div class="col-xs-12 col-sm-3 col-md-3">
+          <label
+            class="col-xs-12 col-md-6 btn btn-light"
+            :disabled="!enabledNextPageButton"
+            @click="fetchNextPage" >
+            Próxima
+          </label>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "pagination",
+
+  props: {
+    totalPages: {
+      required: true,
+    },
+    currentPage: {
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      paginationButtons: [],
+      currentPageData: 1,
```

Comment:
`currentPageMutable`
CurrentPageData sounds like 'the data contained inside the current page'